components/script/dom/gpu.rs

```diff
@@ -139,39 +139,41 @@ impl GPUMethods for GPU {
 }
 
 impl AsyncWGPUListener for GPU {
-    fn handle_response(&self, response: WebGPUResponseResult, promise: &Rc<Promise>) {
+    fn handle_response(&self, response: Option<WebGPUResponseResult>, promise: &Rc<Promise>) {
         match response {
-            Ok(WebGPUResponse::RequestAdapter {
-                adapter_info,
-                adapter_id,
-                features,
-                limits,
-                channel,
-            }) => {
-                let adapter = GPUAdapter::new(
-                    &self.global(),
-                    channel,
-                    DOMString::from(format!(
-                        "{} ({:?})",
-                        adapter_info.name,
-                        adapter_id.0.backend()
-                    )),
-                    Heap::default(),
-                    features,
-                    limits,
+            Some(response) => match response {
+                Ok(WebGPUResponse::RequestAdapter {
                     adapter_info,
                     adapter_id,
-                );
-                promise.resolve_native(&adapter);
-            },
-            Err(e) => {
-                warn!("Could not get GPUAdapter ({:?})", e);
-                promise.resolve_native(&None::<GPUAdapter>);
-            },
-            _ => {
-                warn!("GPU received wrong WebGPUResponse");
-                promise.reject_error(Error::Operation);
+                    features,
+                    limits,
+                    channel,
+                }) => {
+                    let adapter = GPUAdapter::new(
+                        &self.global(),
+                        channel,
+                        DOMString::from(format!(
+                            "{} ({:?})",
+                            adapter_info.name,
+                            adapter_id.0.backend()
+                        )),
+                        Heap::default(),
+                        features,
+                        limits,
+                        adapter_info,
+                        adapter_id,
+                    );
+                    promise.resolve_native(&adapter);
+                },
+                Err(e) => {
+                    unreachable!("Failed to get a response from the constellation ({})", e)
```

Comment:
So I was wrong about this being unreachable, because here we are handling the "inner" `WebGPUResponseResult`, not the receiving of the message containing this result, which is done at https://github.com/servo/servo/blob/c219204084b8f8a747ebd37ec75472c1b7e97411/components/script/dom/gpu.rs#L77. 

Conclusion: this branch should remain ```               
 warn!("Could not get GPUAdapter ({:?})", e);
                promise.resolve_native(&None::<GPUAdapter>);```

and it's the `None` branch that should ```warn!("Couldn't get a response, because WebGPU is disabled ({})", e); promise.resolve_native(&None::<GPUAdapter>);```

Reply:
Got it, thanks for explaining, I addressed other comments.